torch/_inductor/scheduler.py

```diff
@@ -969,10 +988,8 @@ def __init__(
         else:
             self.scheduler = scheduler
             self.snodes = nodes
-            self.node = None
-
-            self.node = None
-            self.users = None
+            self.node = None  # type: ignore[assignment]
```

Comment:
can we make BaseSchedulerNode.node an `Optional` type to avoid this `type: ignore`?